geomesa-convert/geomesa-convert-common/src/main/scala/org/locationtech/geomesa/convert/Transformers.scala

```diff
@@ -0,0 +1,327 @@
+      case l ~ "," ~ r => StrEQ(l, r)
+    }
+    def intEq       = ("intEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntEQ(l, r)
+    }
+    def intLTEq     = ("intLTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntLTEQ(l, r)
+    }
+    def intLT       = ("intLT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntLT(l, r)
+    }
+    def intGTEq     = ("intGTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntGTEQ(l, r)
+    }
+    def intGT       = ("intGT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => IntGT(l, r)
+    }
+    def dEq       = ("dEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DEQ(l, r)
+    }
+    def dLTEq     = ("dLTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DLTEQ(l, r)
+    }
+    def dLT       = ("dLT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DLT(l, r)
+    }
+    def dGTEq     = ("dGTEq" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DGTEQ(l, r)
+    }
+    def dGT       = ("dGT" ~ OPEN_PAREN) ~> (transformExpr ~ "," ~ transformExpr) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => DGT(l, r)
+    }
+    def binaryPred  = strEq | intEq | intLTEq | intLT | intGTEq | intGT | dEq | dLTEq | dLT | dGTEq | dGT
+    def andPred     = ("and" ~ OPEN_PAREN) ~> (pred ~ "," ~ pred) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => And(l, r)
+    }
+    def orPred      = ("or" ~ OPEN_PAREN) ~> (pred ~ "," ~ pred) <~ CLOSE_PAREN ^^ {
+      case l ~ "," ~ r => Or(l, r)
+    }
+    def notPred     = ("not" ~ OPEN_PAREN) ~> pred <~ CLOSE_PAREN ^^ {
+      case pred => Not(pred)
+    }
+    def logicPred = andPred | orPred | notPred
+    def pred: Parser[Predicate] = binaryPred | logicPred
+    def expr = fn | wholeRecord | regexExpr | fieldLookup | column | lit
+    def transformExpr: Parser[Expr] = cast2double | cast2int | expr
+  }
+
+  class EvaluationContext(fieldNameMap: Map[String, Int], var computedFields: Array[Any]) {
+    def indexOf(n: String) = fieldNameMap(n)
+    def lookup(i: Int) = computedFields(i)
+  }
+
+  sealed trait Expr {
+    def eval(args: Any*)(implicit ctx: EvaluationContext): Any
+  }
+
+  sealed trait Lit[T <: Any] extends Expr {
+    def value: T
+    override def eval(args: Any*)(implicit ctx: EvaluationContext): Any = value
+  }
+
+  case class LitString(value: String) extends Lit[String]
+  case class LitInt(value: Integer) extends Lit[Integer]
+  case class LitDouble(value: java.lang.Double) extends Lit[java.lang.Double]
```

Comment:
I might be missing it; should LitLong/long support be added?  In the same vein, should we have float support?